feat: FIT-450: Improve empty state for data manager

ricardoantoniocm opened the pull request:
<!-- Follow Merge -->
This PR was automatically generated via Follow Merge. [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/17166672335)
Please ensure that all linked upstream Pull Requests are merged before proceeding with this one.
- https://github.com/HumanSignal/label-studio-sdk/pull/545
<!-- Follow Merge -->




## Reason for change

This PR addresses the user experience of the data manager's empty state. The previous "blank" state was unhelpful and didn't provide users with clear guidance on how to get started. This change introduces a new, comprehensive empty state component that is more informative.

The new component provides different messaging and actions based on the context:

* **Empty Projects:** Guides Administrators and Managers to import their initial data (via Connecting Cloud Storage or from their computer using the Import modal) and, guides Reviewers and Annotators on what they can do given their role (when applicable). 
* A link to documentation is added that works dynamically depending on whether it's presented on LSO or LSE. It is excluded from whitelabeled environments.
* **Empty Filter Results:** Informs users when their current filters return no results and provides an easy way to clear the filters.

This improvement enhances the user experience by providing a clear path forward, reducing friction for new users, and improving usability for existing users.

The Connect Storage button pushes the user to the Cloud Storages page and automatically opens the Add Source Storage modal, streamlining the process. It also leverages the Import modal for processing any uploaded files.

This PR also introduces improvements to the destructive confirmation dialogs used in the Data Manager, fixing the variants of the buttons as well as implementing more meaningful copy on the title, body text and buttons.

## Screenshots

### Before

Empty State
<img width="1492" height="872" alt="image" src="https://github.com/user-attachments/assets/a1ad585f-1a40-4e40-bc73-51682c45e083" />

No results from filters
<img width="1493" height="971" alt="image" src="https://github.com/user-attachments/assets/a93b7971-20b4-4bbd-aed5-4fc50983d8f1" />

Destructive DM Actions
<img width="490" height="300" alt="image" src="https://github.com/user-attachments/assets/af1e4421-8a69-44f9-aa83-4e68c6e24230" />

### After

Empty State
<img width="1948" height="1116" alt="image" src="https://github.com/user-attachments/assets/bc6b8b4b-1ab1-49cc-be45-a627d96f6365" />

No results from filters
<img width="1315" height="840" alt="image" src="https://github.com/user-attachments/assets/f107076f-0c43-4ebf-998b-65c429b1f397" />

Destructive DM Actions
<img width="725" height="312" alt="image" src="https://github.com/user-attachments/assets/bf425f1c-06f9-404f-bebd-c03ac9082a83" />

### Rollout strategy

* The new `EmptyState` component will be deployed as part of the normal release cycle.
* No feature flag is necessary as this change is a direct improvement to the existing UI.
* The change is fully backward compatible, as it only affects the presentation layer when a dataset is empty.

### Testing

I've conducted the following testing to ensure the new component works as expected:

* **Unit Tests:** Added comprehensive unit tests for the `EmptyState` component, covering all display modes and props.
* **Integration Tests:**
    * Verified that the correct empty state is displayed for a brand new project with no data.
    * Confirmed that the "Empty Filter Results" state appears when applying a filter that yields no results.
    * Tested the functionality of the "Import" and "Clear Filters" buttons.
    * Ensured the component integrates correctly with the main `DataManager` and `Table` components.
    * Verified the component works as expected on different screen sizes to confirm responsiveness.

### Reviewer notes

* Please pay close attention to the visual implementation of the new empty states.
* Verify that the conditional rendering logic correctly differentiates between truly empty projects and empty filtered results.
* Some files may have some automatic linting changes applied, unrelated to the work itself. Those changes should be safe to keep though.


<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

ricardoantoniocm committed:
feat: FIT-450: Improve empty state for data manager

ricardoantoniocm committed:
Implements empty state on Data Manager surfacing Connect Storage, drag and drop and browse to upload functionality.

ricardoantoniocm committed:
Adds docs link.

ricardoantoniocm committed:
Linter fixes.

ricardoantoniocm committed:
Improves confirmation/destructive action dialogs.

ricardoantoniocm committed:
Linter fixes.

ricardoantoniocm committed:
Increases padding for empty state.

ricardoantoniocm committed:
Adds accessibility improvements and tests.

ricardoantoniocm committed:
Improvements.

ricardoantoniocm committed:
Introduces role-based handling for the empty state, including improved results filter empty state. Adds inbox icon.

ricardoantoniocm committed:
Refactors to follow react rules.

ricardoantoniocm committed:
Fix linting issues: convert EmptyState to TypeScript and fix ARIA roles

ricardoantoniocm committed:
Apply auto-formatting fixes

ricardoantoniocm committed:
Fix EmptyState role prop usage in Table component

The EmptyState component was updated to use 'userRole' instead of 'role'
to avoid ARIA role conflicts, but Table.jsx was still passing 'role'.
This caused the Reviewer role to show the default empty state instead
of the role-specific content.

ricardoantoniocm committed:
Fix EmptyState logic: prioritize role-specific states over filter states

- Annotators with auto-distribution now see role-specific empty state
  even when filters are applied, instead of generic filter message
- Filter empty state now only applies to Owners/Admins/Managers
- Fixed logic order: role-specific checks first, then filter checks
- Updated both EmptyState.tsx files to maintain consistency
- Cleared caches to ensure fresh component loading

ricardoantoniocm committed:
Add debug logging for hasFilters calculation

This will help identify why hasFilters is incorrectly true when no filters are applied.
Check console for 'DEBUG hasFilters' messages to see the filtersApplied value.

ricardoantoniocm committed:
Revert "Add debug logging for hasFilters calculation"

This reverts commit 9daee6d50323021cb473bda420a8f98952080470.

ricardoantoniocm committed:
Removes legacy check for filters. Fixes order of filters empty state.

ricardoantoniocm committed:
Fix tests.

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16946783474)
> Successfully merged:  delete mode 100644 web/libs/editor/src/regions/ImageRegion.js

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-450/improve-empty-state-data-manager'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16946783474

ricardoantoniocm committed:
Adds data-testids to buttons on destructive/confirmation data manager dialogs.

ricardoantoniocm committed:
Adds data-testids to Import and Export buttons on Data Manager.

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16991079283)
> Successfully merged:  create mode 100644 web/libs/editor/tests/integration/e2e/control_tags/text-with-dual-taxonomy.cy.ts

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-450/improve-empty-state-data-manager'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16991079283

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17041289556)
> Successfully merged:  create mode 100644 web/libs/ui/src/shad/components/ui/accordion.tsx

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-450/improve-empty-state-data-manager'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17041289556

ricardoantoniocm committed:
Remove interactive upload in empty state. Removes leftover empty state file and test file.

ricardoantoniocm committed:
Reverts changes to Import.

ricardoantoniocm committed:
Revert changes to Data Manager page related to processing import from the removed empty state functionality.

ricardoantoniocm committed:
Refactors comment on StorageSet.

ricardoantoniocm committed:
Refactors EmptyState component to reduce repeated code.

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17051876534

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-fit-450/improve-empty-state-data-manager'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17051876534

ricardoantoniocm committed:
Slightly copy adjustment to avoid repetition.

nass600 reviewed (approved):
@ricardoantoniocm Looks good. I believe this EmptyState could be promoted to the Design System as we have it spread all over the app

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17100417382)
> Successfully merged:  47 files changed, 1763 insertions(+), 97 deletions(-)

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-450/improve-empty-state-data-manager'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17100417382

ricardoantoniocm committed:
Fixes typo.

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17164304510)
> Successfully merged:  67 files changed, 3812 insertions(+), 1028 deletions(-)

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-450/improve-empty-state-data-manager'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17164304510

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17165254274

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17166412632

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-fit-450/improve-empty-state-data-manager'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17166412632

ricardoantoniocm commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17166672335)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17166652170

ricardoantoniocm committed:
Updates test with correct string.

ricardoantoniocm committed:
Linter fixes.

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio-enterprise/actions/runs/17167243443
